source/Meadow.Core/Update/UpdateService.cs

```diff
@@ -572,6 +580,49 @@ private void DisplayTree(DirectoryInfo di)
         }
     }
 
+    // In order to not require the use of a library, this was taken from
+    // RFC7515, Appendix C: https://datatracker.ietf.org/doc/html/rfc7515#appendix-C
+    private string base64UrlDecode(string arg)
+    {
+        string s = arg;
+        s = s.Replace('-', '+'); // 62nd char of encoding
+        s = s.Replace('_', '/'); // 63rd char of encoding
+        switch (s.Length % 4) // Pad with trailing '='s
+        {
+            case 0: break; // No pad chars in this case
+            case 2: s += "=="; break; // Two pad chars
+            case 3: s += "="; break; // One pad char
+            default:
+                throw new Exception("Illegal base64url string!");
+        }
+
+        var bytes = Convert.FromBase64String(s);
+        return Encoding.UTF8.GetString(bytes);
+    }
+
+    private JsonWebTokenPayload? GetJsonWebTokenPayload(string? jwt)
+    {
+        if (string.IsNullOrWhiteSpace(jwt))
```

Comment:
totally minor, but I think perf is better if you use `ArgumentNullException.ThrowIfNullOrEmpty()`

Reply:
From what I can tell `netstandard2.1` does not have `ArgumentNullException.ThrowIfNull()` nor `ArgumentException.ThrowIfNullOrEmpty()`.